types/node/fs.d.ts

```diff
@@ -108,38 +108,52 @@ declare module "fs" {
-
-        on(event: string, listener: (...args: any[]) => void): this;
-        on(event: "change", listener: (eventType: string, filename: string | Buffer) => void): this;
-        on(event: "error", listener: (error: Error) => void): this;
-        on(event: "close", listener: () => void): this;
-
-        once(event: string, listener: (...args: any[]) => void): this;
-        once(event: "change", listener: (eventType: string, filename: string | Buffer) => void): this;
-        once(event: "error", listener: (error: Error) => void): this;
-        once(event: "close", listener: () => void): this;
-
-        prependListener(event: string, listener: (...args: any[]) => void): this;
-        prependListener(event: "change", listener: (eventType: string, filename: string | Buffer) => void): this;
-        prependListener(event: "error", listener: (error: Error) => void): this;
-        prependListener(event: "close", listener: () => void): this;
-
-        prependOnceListener(event: string, listener: (...args: any[]) => void): this;
-        prependOnceListener(event: "change", listener: (eventType: string, filename: string | Buffer) => void): this;
-        prependOnceListener(event: "error", listener: (error: Error) => void): this;
-        prependOnceListener(event: "close", listener: () => void): this;
+        addListener<K extends keyof FSWatcherEventMap>(event: K, listener: FSWatcherEventMap[K]): this;
+        addListener(event: string | symbol, listener: (...args: any[]) => void): this;
+
+        emit<K extends keyof FSWatcherEventMap>(event: K, ...args: FSWatcherEventMap[K] extends (...args: infer P) => any ? P : never): boolean;
+        emit(event: string | symbol, ...args: any[]): boolean;
+
+        on<K extends keyof FSWatcherEventMap>(event: K, listener: FSWatcherEventMap[K]): this;
+        on(event: string | symbol, listener: (...args: any[]) => void): this;
+
+        once<K extends keyof FSWatcherEventMap>(event: K, listener: FSWatcherEventMap[K]): this;
+        once(event: string | symbol, listener: (...args: any[]) => void): this;
+
+        prependListener<K extends keyof FSWatcherEventMap>(event: K, listener: FSWatcherEventMap[K]): this;
+        prependListener(event: string | symbol, listener: (...args: any[]) => void): this;
+
+        prependOnceListener<K extends keyof FSWatcherEventMap>(event: K, listener: FSWatcherEventMap[K]): this;
+        prependOnceListener(event: string | symbol, listener: (...args: any[]) => void): this;
+
+        removeListener<K extends keyof FSWatcherEventMap>(event: K, listener: FSWatcherEventMap[K]): this;
+        removeListener(event: string | symbol, listener: (...args: any[]) => void): this;
+
+        off<K extends keyof FSWatcherEventMap>(event: K, listener: FSWatcherEventMap[K]): this;
+        off(event: string | symbol, listener: (...args: any[]) => void): this;
+
+		listeners<K extends keyof FSWatcherEventMap>(event: K): Array<FSWatcherEventMap[K]>;
```

Comment:
<!--
  1 failure:  The root .editorc...
  0 warning: 
  
  
  DangerID: danger-id-Danger;
  File: types/node/fs.d.ts;
  Line: 150;
-->
- :no_entry_sign: The root .editorconfig style specifies spaces for whitespace. Please use spaces in new or changed types.